test/e2e/host_path.go

```diff
@@ -0,0 +1,141 @@
+
+import (
+	"fmt"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/latest"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"os"
+	"path"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+//TODO : Consolodate this code with the code for emptyDir.
+//This will require some smart.
+var _ = Describe("hostDir", func() {
+	var (
+		c         *client.Client
+		namespace *api.Namespace
+	)
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Building a namespace api object")
+		namespace, err = createTestingNS("hostdir", c)
+		Expect(err).NotTo(HaveOccurred())
+
+		//cleanup before running the test.
+		_ = os.Remove("/tmp/test-file")
+	})
+
+	AfterEach(func() {
+		By(fmt.Sprintf("Destroying namespace for this suite %v", namespace.Name))
+		if err := c.Namespaces().Delete(namespace.Name); err != nil {
+			Failf("Couldn't delete ns %s", err)
+		}
+	})
+
+	It("MOD volume on tmpfs should have the correct mode", func() {
+		volumePath := "/test-volume"
+		source := &api.HostPathVolumeSource{
+			Path: "/tmp",
+		}
+		pod := testPodWithHostVol(volumePath, source)
+
+		pod.Spec.Containers[0].Args = []string{
+			fmt.Sprintf("--fs_type=%v", volumePath),
+			fmt.Sprintf("--file_mode=%v", volumePath),
+		}
+		testContainerOutputInNamespace("emptydir r/w on tmpfs", c, pod, []string{
+			"mode of file \"/test-volume\": dtrwxrwxrwx", // we expect the sticky bit (mode flag t) to be set for the dir
+		},
+			namespace.Name)
+	})
+
+	It("should support r/w on tmpfs", func() {
+		volumePath := "/test-volume"
+		filePath := path.Join(volumePath, "test-file")
+		source := &api.HostPathVolumeSource{
+			Path: "/tmp",
+		}
+		pod := testPodWithHostVol(volumePath, source)
+
+		pod.Spec.Containers[0].Args = []string{
+			fmt.Sprintf("--fs_type=%v", volumePath),
+			fmt.Sprintf("--rw_new_file=%v", filePath),
+			fmt.Sprintf("--file_mode=%v", filePath),
+		}
+		testContainerOutputInNamespace("emptydir r/w on tmpfs", c, pod, []string{
+			"mode of file \"/test-volume/test-file\": -rw-r--r--",
+			"content of file \"/test-volume/test-file\": mount-tester new file",
+		}, namespace.Name,
+		)
+	})
+})
+
+//These constants are borrowed from the other test.
+//const containerName = "test-container"
+//const volumeName = "test-volume"
+
+func mount(source *api.HostPathVolumeSource) []api.Volume {
+	return []api.Volume{
+		{
+			Name: volumeName,
+			VolumeSource: api.VolumeSource{
+				HostPath: source,
+			},
+		},
+	}
+}
+
+//TODO: To merge this with the emptyDir tests, we can make source a lambda.
+func testPodWithHostVol(path string, source *api.HostPathVolumeSource) *api.Pod {
+	podName := "pod-" + string(util.NewUUID())
+
+	return &api.Pod{
+		TypeMeta: api.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: latest.Version,
+		},
+		ObjectMeta: api.ObjectMeta{
+			Name: podName,
```

Comment:
Its set at line 71, in the `testContainerOutputInNamespace`.  The pod is created in line 71, this guy just creates the data structure. but good catch actually i almost thought you were right because its a little confusing :)